controller/input.h

```diff
@@ -91,7 +93,7 @@ class InputDevice
     InterruptibleReader eventReader;
     std::thread eventThread;
 
-    ff_effect effect = {};
+    ff_effect effects[INPUT_MAX_FF_EFFECTS] = {};
     uint16_t effectGain = 0xffff;
     FeedbackReceived feedbackReceived;
```

Comment:
Look, this is the problematic code: If you write out-of-bounds, you can overwrite the function pointer in `feedbackReceived` - thus provoking a jump to unknown code which the hacker could also conveniently pass through out-of-bounds writes.